types/api.go

```diff
@@ -568,16 +568,30 @@ type ApiValidatorEth1Response struct {
 }
 
 type ApiValidatorIncomeHistoryResponse struct {
-	Income struct {
-		AttestationSourceReward uint64 `json:"attestation_source_reward"`
-		AttestationTargetReward uint64 `json:"attestation_target_reward"`
-		AttestationHeadReward   uint64 `json:"attestation_head_reward"`
-	} `json:"income"`
-	Epoch          uint64    `json:"epoch"`
-	ValidatorIndex uint64    `json:"validatorindex"`
-	Week           uint64    `json:"week"`
-	WeekStart      time.Time `json:"week_start"`
-	WeekEnd        time.Time `json:"week_end"`
+	Income         *ApiValidatorIncomeHistory `json:"income"`
+	Epoch          uint64                     `json:"epoch"`
+	ValidatorIndex uint64                     `json:"validatorindex"`
+	Week           uint64                     `json:"week"`
+	WeekStart      time.Time                  `json:"week_start"`
+	WeekEnd        time.Time                  `json:"week_end"`
+}
+
+type ApiValidatorIncomeHistory struct {
+	AttestationSourceReward            uint64 `json:"attestation_source_reward,omitempty"`
+	AttestationSourcePenalty           uint64 `json:"attestation_source_penalty,omitempty"`
+	AttestationTargetReward            uint64 `json:"attestation_target_reward,omitempty"`
+	AttestationTargetPenalty           uint64 `json:"attestation_target_penalty,omitempty"`
+	AttestationHeadReward              uint64 `json:"attestation_head_reward,omitempty"`
+	FinalityDelayPenalty               uint64 `json:"finality_delay_penalty,omitempty"`
+	ProposerSlashingInclusionReward    uint64 `json:"proposer_slashing_inclusion_reward,omitempty"`
+	ProposerAttestationInclusionReward uint64 `json:"proposer_attestation_inclusion_reward,omitempty"`
+	ProposerSyncInclusionReward        uint64 `json:"proposer_sync_inclusion_reward,omitempty"`
+	SyncCommitteeReward                uint64 `json:"sync_committee_reward,omitempty"`
+	SyncCommitteePenalty               uint64 `json:"sync_committee_penalty,omitempty"`
+	SlashingReward                     uint64 `json:"slashing_reward,omitempty"`
+	SlashingPenalty                    uint64 `json:"slashing_penalty,omitempty"`
+	TxFeeRewardWei                     string `json:"tx_fee_reward_wei,omitempty"`
```

Comment:
Why would this be a string and not, for example, a big.Int?
Then again, this would break the API as it was a []byte before. Thoughts?

If required: Could you please discuss this topic internally? We may want to add a new field, for example.

Reply:
When you display a bigint like 27194623959527110306 as a number it is possible, depending on the way you display it, for it to get converted the wrong way to e.g. 27194623959527110000.
Returning a string prevents this.

Reply:
Now that I think about it, we cannot return a big.Int anyways as this would break the API.